In [1]:
import pandas as pd
import seaborn as sns

%matplotlib inline

In [2]:
df = pd.read_csv("311.csv")

In [3]:
df

Unnamed: 0  Unique Key            Created Date  \
0                 0    45531130  02/02/2020 06:09:17 AM   
1                 1    45529784  02/02/2020 02:15:24 PM   
2                 2    45527528  02/02/2020 02:27:41 AM   
3                 3    45530329  02/02/2020 12:13:18 PM   
4                 4    45528814  02/02/2020 01:59:44 PM   
...             ...         ...                     ...   
6019838     6019838    44063685  10/15/2019 10:35:45 AM   
6019839     6019839    44063692  10/15/2019 06:09:28 AM   
6019840     6019840    44063724  10/15/2019 10:54:45 AM   
6019841     6019841    44063726  10/15/2019 02:07:34 PM   
6019842     6019842    44063737  10/15/2019 11:29:01 AM   

                    Closed Date        Complaint Type         Location Type  \
0                           NaN        HEAT/HOT WATER  RESIDENTIAL BUILDING   
1                           NaN  UNSANITARY CONDITION  RESIDENTIAL BUILDING   
2                           NaN        HEAT/HOT WATER  RESIDENTIAL BUILDING   
3                           NaN        HEAT/HOT WATER  RESIDENTIAL BUILDING   
4                           NaN             APPLIANCE  RESIDENTIAL BUILDING   
...                         ...                   ...                   ...   
6019838  10/16/2019 05:55:02 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019839  10/15/2019 04:12:58 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019840  10/16/2019 06:14:29 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019841  10/16/2019 07:10:19 PM        HEAT/HOT WATER  RESIDENTIAL BUILDING   
6019842  10/15/2019 09:19:45 PM         PAINT/PLASTER  RESIDENTIAL BUILDING   

         Incident Zip         Incident Address         Street Name  \
0             10019.0     426 WEST   52 STREET    WEST   52 STREET   
1             11204.0           1751 67 STREET           67 STREET   
2             11372.0          87-15 37 AVENUE           37 AVENUE   
3             10458.0  2405 SOUTHERN BOULEVARD  SOUTHERN BOULEVARD   
4             11209.0            223 78 STREET           78 STREET   
...               ...                      ...                 ...   
6019838       10029.0     327 EAST  108 STREET    EAST  108 STREET   
6019839       10461.0  3555 BRUCKNER BOULEVARD  BRUCKNER BOULEVARD   
6019840       10034.0       165 SHERMAN AVENUE      SHERMAN AVENUE   
6019841       10467.0    66 WEST GUN HILL ROAD  WEST GUN HILL ROAD   
6019842       10009.0             143 AVENUE D            AVENUE D   

        Address Type             City  Status  \
0            ADDRESS         NEW YORK    Open   
1            ADDRESS         BROOKLYN    Open   
2            ADDRESS  Jackson Heights    Open   
3            ADDRESS            BRONX    Open   
4            ADDRESS         BROOKLYN    Open   
...              ...              ...     ...   
6019838      ADDRESS         NEW YORK  Closed   
6019839      ADDRESS            BRONX  Closed   
6019840      ADDRESS         NEW YORK  Closed   
6019841      ADDRESS            BRONX  Closed   
6019842      ADDRESS         NEW YORK  Closed   

                                    Resolution Description    Borough  \
0        The following complaint conditions are still o...  MANHATTAN   
1        The following complaint conditions are still o...   BROOKLYN   
2        The following complaint conditions are still o...     QUEENS   
3        The following complaint conditions are still o...      BRONX   
4        The following complaint conditions are still o...   BROOKLYN   
...                                                    ...        ...   
6019838  The complaint you filed is a duplicate of a co...  MANHATTAN   
6019839  The Department of Housing Preservation and Dev...      BRONX   
6019840  The complaint you filed is a duplicate of a co...  MANHATTAN   
6019841  The Department of Housing Preservation and Dev...      BRONX   
6019842  The Department of Housing Preservation and Dev...  MANHATTAN   

          Latitude  Longitude  
0        40.

### Cardinality check

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6019843 entries, 0 to 6019842
Data columns (total 16 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   Unnamed: 0              int64  
 1   Unique Key              int64  
 2   Created Date            object 
 3   Closed Date             object 
 4   Complaint Type          object 
 5   Location Type           object 
 6   Incident Zip            float64
 7   Incident Address        object 
 8   Street Name             object 
 9   Address Type            object 
 10  City                    object 
 11  Status                  object 
 12  Resolution Description  object 
 13  Borough                 object 
 14  Latitude                float64
 15  Longitude               float64
dtypes: float64(3), int64(2), object(11)
memory usage: 734.8+ MB


### Checking Address feature for cardinality since it's categorical

In [5]:
df["Incident Address"].describe()

count             5967018
unique             182600
top       34 ARDEN STREET
freq                14298
Name: Incident Address, dtype: object

In [6]:
df["Street Name"].describe()

count             5967018
unique               6825
top       GRAND CONCOURSE
freq                92450
Name: Street Name, dtype: object

In [7]:
df["Address Type"].describe()

count     5935078
unique          1
top       ADDRESS
freq      5935078
Name: Address Type, dtype: object

### Observation: Total count is 5967018 but unique count (cardinality) is 182600